std/random.d

```diff
@@ -1183,6 +1183,130 @@ A singleton instance of the default random number generator
 }
 
 /**
+A random bool generator that has the underlying random integral
+generation batched.
+ */
+struct RandomBool(Rng = Random)
+    if (isUniformRNG!Rng || isUniformRNG!(typeof(*Rng.init)))
+{
+    private Rng parent;
+    private uint mask;
+    private uint accumulator;
+
+    /// Always `false`
+    enum bool empty = false;
+
+    static if (isPointer!Rng)
+    {
+        this(Rng rng) @safe
+        {
+            parent = rng;
+            initPrivates();
+        }
+    }
+    else
+    {
+        /**
+          If Rng is not a pointer, we want to pass it by ref
+          in order to avoid two passByValues
+          */
+        this(ref Rng rng) @safe
+        {
+            parent = rng;
+            initPrivates();
+        }
+    }
+
+    private void initPrivates()
+    {
+        // Helper function to avoid code duplication in constructor
+        mask = 1;
+        accumulator = uniform!uint(parent);
+    }
+
+    @property bool front() @safe pure nothrow @nogc
```

Comment:
drop `@safe pure nothrow @nogc`, they will be deduced depending on `Rng`. There is a guideline somewhere - if you specify these here, you limit yourself to `Rng`s that are safe, pure, nothrow etc. and you don't want that.